llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -1084,6 +1084,17 @@ bool ScalarizerVisitor::visitExtractValueInst(ExtractValueInst &EVI) {
   ValueVector Res;
   if (!isStructOfMatchingFixedVectors(OpTy))
     return false;
+  if (CallInst *CI = dyn_cast<CallInst>(Op)) {
```

Comment:
Can you just handle `IntrinsicInst` here, instead of manually checking for a known callee and looking up the id?

Reply:
Still need to lookup the id, but yes this could be `IntrinsicInst`.